test/datasource.test.js

```diff
@@ -545,6 +545,33 @@ describe('DataSource', function() {
       ds.connector.should.equal(connector);
     });
   });
+
+  describe('getMaxOfflineRequests', () => {
+    let ds;
+    beforeEach(() => ds = new DataSource('ds', {connector: 'memory'}));
+
+    it('sets the default maximum number of event listeners to 16', () => {
+      ds.getMaxOfflineRequests().should.be.eql(16);
+    });
+
+    it('uses the provided max number of listeners even if it is less than 16', () => {
+      ds.settings.maxOfflineRequests = 15;
+      ds.getMaxOfflineRequests().should.be.eql(15);
+    });
+
+    it('uses provided number of listeners if it is greater than 16', () => {
```

Comment:
nitpick: I think the value of `settings.maxOfflineRequests` doesn't matter, so 15 and 17 looks like a dup test to me. (or did I miss something?)

Reply:
It was because before we were only assigning it if it was greater, but now that that's removed I can remove this test, thanks!